lib/quality/quality_score.rb

```diff
@@ -0,0 +1,103 @@
+class QualityScore
+  attr_reader :reasons
+
+  def initialize(dataset)
+    @dataset = dataset
+    @reasons = []
+  end
+
+  # Obtain a quality score for the dataset out of a maximum of 100.
+  # Points are removed for each failing feature of the dataset.
+  def score
+    value = 100
+
+    methods = QualityScore.instance_methods.grep(/.*_score/)
+    methods.each do | f |
+      value -= self.send(f)
+    end
+
+    value
+  end
+
+  def frequency_score
+    if @dataset.frequency.blank?
+      @reasons << "This dataset has no update frequency set"
+      10
+    else
+      0
+    end
+  end
+
+  def documentation_score
+    if @dataset.docs.count() == 0
```

Comment:
Not sure you need the brackets here or in the other methods ie...`@dataset.docs.count == 0`, `@dataset.links.all` etc